cmd/git_utils.go

```diff
@@ -0,0 +1,215 @@
+// Copyright © 2019 IBM Corporation and others.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cmd
+
+import (
+	"encoding/json"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type CommitInfo struct {
+	Author string
+	SHA    string
+	Date   string
+	URL    string
+}
+
+type GitInfo struct {
+	Branch      string
+	Upstream    string
+	RemoteURL   string
+	Changes     []string
+	ChangesMade bool
+	Commit      CommitInfo
+}
+
+const trimChars = "' \r\n"
+
+func stringBefore(value string, searchValue string) string {
+	// Get substring before a string.
+
+	gitURLElements := strings.Split(value, searchValue)
+	if len(gitURLElements) == 0 {
+		return ""
+	}
+	return gitURLElements[0]
+
+}
+
+func stringAfter(value string, searchValue string) string {
+	// Get substring after a string.
+	position := strings.LastIndex(value, searchValue)
+	if position == -1 {
+		return ""
+	}
+	adjustedPosition := position + len(searchValue)
+	if adjustedPosition >= len(value) {
+		return ""
+	}
+	return value[adjustedPosition:]
+}
+func stringBetween(value string, pre string, post string) string {
+	// Get substring between two strings.
+	positionBegin := strings.Index(value, pre)
+	if positionBegin == -1 {
+		return ""
+	}
+	positionEnd := strings.Index(value, post)
+	if positionEnd == -1 {
+		return ""
+	}
+	positionBeginAdjusted := positionBegin + len(pre)
+	if positionBeginAdjusted >= positionEnd {
+		return ""
+	}
+	return value[positionBeginAdjusted:positionEnd]
+}
+
+//RunGitFindBranc issues git status
+func GetGitInfo(dryrun bool) (GitInfo, error) {
+	var gitInfo GitInfo
+	version, vErr := RunGitVersion(false)
+	if vErr != nil {
+		return gitInfo, vErr
+	}
+	if version == "" {
+		return gitInfo, errors.Errorf("git does not appear to be available")
+	}
+
+	Debug.log("git version: ", version)
+
+	kargs := []string{"status", "-sb"}
+
+	output, gitErr := RunGit(kargs, dryrun)
+	if gitErr != nil {
+		return gitInfo, gitErr
+	}
+
+	lineSeparator := "\n"
+	if runtime.GOOS == "windows" {
+		lineSeparator = "\r\n"
+	}
+	outputLines := strings.Split(output, lineSeparator)
+
+	var changeLines []string
+	const noCommits = "## No commits yet on "
+	const branchPrefix = "## "
+	const branchSeparatorString = "..."
```

Comment:
I tested this command locally, and did not get a `branchSeparatorString` after my branch name. 
Instead I get just 
![image](https://user-images.githubusercontent.com/30461857/66398934-8e3ac480-e9d6-11e9-92ed-6c36ccd9bce7.png)
So the branch field in GitInfo is not set correctly. Is the separator needed when you have multiple branches?

(Note: git.go is the local version of this file)

Reply:
what do you get when you do git remote -v out of curiosity?  What about git config?

Reply:
It's ok I can reproduce it.